File: Assignment31_3.c
#include <stdio.h>

///////////////////////////////////////////////////////////////
//  Function Name : Pattern
//  Description   : Display diagonal based pattern
//  Input         : Integer 
//  Output        : $   *   *   *   *
//                  #   $   *   *   *
//                  #   #   $   *   *
//                  #   #   #   $   *
//                  #   #   #   #   $
//
//  Author        : Raj Samir Jadhav
//  Date          : 25/11/2025
///////////////////////////////////////////////////////////////


void Pattern(int iRow, int iCol)
{
    int i = 0, j = 0;

    for(i = 1; i <= iRow; i++)
    {
        for(j = 1; j <= iCol; j++)
        {
            if(i == j)                 // Diagonal
            {
                printf("$\t");
            }
            else if(j < i)             // Left side
            {
                printf("#\t");
            }
            else                       // Right side
            {
                printf("*\t");
            }
        }
        printf("\n");
    }
}

int main()
{
    int iValue1 = 0, iValue2 = 0;

    printf("Enter number of rows and columns : \n");
    scanf("%d %d", &iValue1, &iValue2);

    Pattern(iValue1, iValue2);

    return 0;
}

File: Assignment31_3.java
import java.util.Scanner;

///////////////////////////////////////////////////////////////
//  Function Name : Pattern
//  Description   : Display diagonal based pattern
//  Input         : Integer 
//  Output        : $   *   *   *   *
//                  #   $   *   *   *
//                  #   #   $   *   *
//                  #   #   #   $   *
//                  #   #   #   #   $
//
//  Author        : Raj Samir Jadhav
//  Date          : 25/11/2025
///////////////////////////////////////////////////////////////

class Pattern
{
    public void Display(int iRow, int iCol)
    {
        for(int i = 1; i <= iRow; i++)
        {
            for(int j = 1; j <= iCol; j++)
            {
                if(i == j)               // Diagonal
                {
                    System.out.print("$\t");
                }
                else if(j < i)           // Left side
                {
                    System.out.print("#\t");
                }
                else                     // Right side
                {
                    System.out.print("*\t");
                }
            }
            System.out.println();
        }
    }
}

class Assignment31_3
{
    public static void main(String[] args)
    {
        Scanner sobj = new Scanner(System.in);

        int iValue1 = 0, iValue2 = 0;

        System.out.println("Enter number of rows and columns : ");
        iValue1 = sobj.nextInt();
        iValue2 = sobj.nextInt();

        Pattern pobj = new Pattern();
        pobj.Display(iValue1, iValue2);
    }
}
